code/modules/mining/mine_turfs.dm

```diff
@@ -689,3 +737,42 @@ var/list/mining_overlay_cache = list()
 		if(initialized)
 			UpdateMineral()
 	update_icon()
+
+// The value of range is turf-radius.
+// digsite_type allows you to specify what type of digsite is spawned. code/modules/xenoarchaeology/finds/finds_defines.dm
+// allow_anomalies is the probability, if any, of spawning an artifact in a turf, in the range. Garden and Animal digsites overwrite this probability.
+/turf/simulated/mineral/proc/make_digsite(var/range = 0, var/digsite_type , var/allow_anomalies = 5)
+	var/digsite = digsite_type ? digsite_type : get_random_digsite_type()
+
+	for(var/turf/simulated/mineral/T in range(range,src))
+		if(LAZYLEN(T.finds))
+			continue
+
+		if(isnull(T.geologic_data))
+			T.geologic_data = new /datum/geosample(T)
+		
+		if(isnull(T.finds))
```

Comment:
This won't populate it with anything if `finds` is an empty `list()`, is that intentional?